cucumber-selenium-tests/src/test/java/com/skillstorm/StepDefinitions/GenericStepDefinitions.java

```diff
@@ -1,19 +1,88 @@
+        this.driver = driver;
+    }
+
+    /*
+     * General use definition for page loading step. Uses navigator class in Utilities package
+     * -------------------------------------------------------------------------------------------------------------
+     * Parameters:
+     * pageName - the page name you'd want to navigate to. It must match Navigator.java naming convention.
+     * -------------------------------------------------------------------------------------------------------------
+     * If you haven't already, you must redefine the method for navigation for your respective page and set the URL for your page as well.
+     */
+
+    @Given("I am on the {string} page")
+    public void iAmOnThePage(String pageName){
+        navigator.navigateTo(pageName);
+    }
+
+
+    /*
+     * General use definition for button click step.
+     * -------------------------------------------------------------------------------------------------------------
+     * Parameters:
+     * buttonId - pass in the buttonId for the button you'd like to press.
+     * -------------------------------------------------------------------------------------------------------------
+     */
+    
+    
+    @When("I click the {string} button")
+    public void iClickTheButton(String buttonId){
+        WebElement button = driver.findElement(By.id(buttonId));
+        button.click();
+    }
+
+    /*
+     * General use definition for selecting an option from a dropdown
+     * -------------------------------------------------------------------------------------------------------------
+     * Parameters:
+     * option - the text for the option you'd like to select
+     * dropdownId - the ID of the dropdown element you want to select for.
+     * -------------------------------------------------------------------------------------------------------------
+     * Notes:
+     * This assumes that each option is dynamically loaded, so it uses the actual text of the option rather than any specific ID
+     * This also assumes your dropdown element has an associated ID
+     */
+
+    @When("I select {string} from the {string} selection")
+    public void iSelectFromTheSelection(String option, String dropdownId){
+
+        //Wait until the dropdown is fully loaded and enabled and set it to variable: dropdown.
+        WebElement dropdown = wait.until(ExpectedConditions.visibilityOfElementLocated(By.id(dropdownId)));
+
+        //Select the correct option in the dropdown, ensure desired option is available
+        Select select = new Select(dropdown);
+        wait.until(ExpectedConditions.textToBePresentInElement(dropdown, option));
+        select.selectByVisibleText(option);
+
+    }
+
+    /*
+     * General use definition for page redirection. Works exactly the same as iAmOnThePage()
+     */
+    @Then("I am redirected to the {string} page")
+    public void iAmRedirectedToThePage(String pageName){
+        navigator.navigateTo(pageName);
```

Comment:
I feel this step, since it is a @Then statement, is supposed to CHECK if we made to a certain page. It is not supposed to take us to that page.

Reply:
I believe this is just missing an assertion to check we are on pageName. We should still leave the navigateTo() in this method, that logic fits into the scenario of "I am redirected to the {string} page".